src/main/scala/mesosphere/marathon/core/storage/zookeeper/PersistenceStore.scala

```diff
@@ -79,6 +81,14 @@ trait PersistenceStore {
     * and submits them. An exception is thrown if one of the operations fail.
     */
   def transaction(operations: Seq[StoreOp]): Future[Done]
+
+  /**
+    * Create a node if none exists with the given path. Return the path of the node. This operation is atomic.
```

Comment:
Should we provide more feedback? Maybe I want to know if it *did* exist.

Reply:
A similar _putIfAbsent_ method in the _ConcurrentHashMap_ [returns](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html#putIfAbsent(K,%20V)) value associated with the key if it exists or _null_ otherwise. However, in our case, only _"key"_ is being created so I'm not sure if this differentiation provides any additional value for the user. At the very least it will make the method signature inconsistent with other storage methods.